Storage/Processor/Config.php

```diff
@@ -39,19 +46,36 @@ public function __construct(StorageManager $storageManager, string $processor, s
         $this->assetHash = $assetHash;
         $this->options = $this->resolve($options);
         $this->configHash = $configHash;
+        $this->setCacheKeyAndFilename();
+        $this->setCacheableResult();
+
+        unset($options[EmsFields::CONTENT_PUBLISHED_DATETIME_FIELD]); //the published date can't invalidate the cache as it'sbased on the config hash now.
+    }
+
+    private function setCacheKeyAndFilename()
+    {
+        $this->cacheKey = $this->assetHash . '_' . $this->configHash;
         $this->filename = null;
 
-        if (!empty($this->getFileNames())) {
-            foreach ($this->getFileNames() as $filename) {
-                if (is_file($filename)) {
-                    $this->filename = $filename;
-                    $this->assetHash = $this->storageManager->computeFileHash($filename);
-                    break;
-                }
+        if ($this->getFileNames() === null) {
+            return;
+        }
+
+        foreach ($this->getFileNames() as $filename) {
+            if (is_file($filename)) {
+                $this->filename = $filename;
+                $this->cacheKey .= '_' . $this->storageManager->computeFileHash($filename);
+                break;
             }
         }
 
-        unset($options[EmsFields::CONTENT_PUBLISHED_DATETIME_FIELD]); //the published date can't invalidate the cache as it'sbased on the config hash now.
+        if ($this->filename === null) {
+            throw new NotFoundHttpException('File not found');
+        }
+
+        if (\in_array($this->options[EmsFields::ASSET_CONFIG_MIME_TYPE], ['application/octet-stream', 'application/bin', ''])) {
```

Comment:
What if I pass a .txt file and define application/json in the options array?

Reply:
As a specific mime type is defined “application/json” it won’t be overwritten by a guess one. Only default mime types (from controller “application/bin” or Config default value “application/octet-stream”) or  an empty string are overwritten

Reply:
Can the if test be a function in the config object? $config->hasDefaultMimeType() ?

Reply:
Refactored has suggested